release-notes/swan-lake-GA-release-note.md

```diff
@@ -27,10 +27,237 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
 
 #### New Features
 
-#### Improvements
+##### Support to Refer to Parameters and Local Variables in Object Constructor Expressions
+
+Object constructor expressions can now refer to parameters and local variables. This feature is also available when using an object constructor expression with the `service` qualifier to create a service object. These references can be in field initializers and object methods including the `init` method.
+
+```ballerina
+type Person object {
+   string name;
+   string code;
+  
+   function getRegistrationId(string prefix) returns string;
+};
+ 
+function createPerson(string name, int id, string department) returns Person|error {
+   string codeId = check getCode(id);
+ 
+   Person person = object {
+       string name;
+       string code = codeId; // Refers to the `codeId` local variable.
+ 
+       function init() {
+           self.name = name; // Refers to the `name` parameter.
+       }
+ 
+       function getRegistrationId(string prefix) returns string {
+           string regId = department + ":" + self.code; // Refers to the `department` parameter.
+ 
+           if (prefix.length() > 0) {
+               return prefix + "-" + regId;
+           }
+ 
+           return regId;
+       }
+   };
+ 
+   return person;
+}
+ 
+function getCode(int id) returns string|error {
+   if id < 0 {
+       return error("Invalid ID");
+   }
+   return id.toString();
+}
+```
 
 #### Bug Fixes
 
+- A bug which resulted in constants being resolved with incorrect types in certain scenarios has been fixed. The type of constants is now the intersection of readonly and the singleton type containing just the shape of the value named by the constant.
+
+```ballerina
+// Previously, the type of X was `int` and now it is singleton `2`.
+const int X = 1 + 1;
+ 
+public function main() {
+   // Now results in an error.
+   X y = 3;
+}
+```
+
+```ballerina
+// Previously, the type of X was `map<int>` and now it is the intersection of readonly
+// and the singleton type containing just the mapping value `{a: 1}`.
+const map<int> X = {a: 1};
+ 
+public function main() {
+   // Now, this results in an error.
+   X y = {a: 2};
+}
+```
```

Comment:
````suggestion
  ```ballerina
  // Previously, the type of X was `map<int>` and now it is the intersection of readonly
  // and the singleton type containing just the mapping value `{a: 1}`.
  const map<int> X = {a: 1};
   
  public function main() {
     // Now, this results in an error.
     X y = {a: 2};
  }
  ```
````